### ISSUE TITLE
Cannot parse comma separated identity specification elements

### ISSUE THREAD

**Issue by rossetti** [jOOQ/jOOQ#7627]
### SQL translator incorrectly translates this query:

```sql
CREATE TABLE simulation_run 
  ( 
     id                         INTEGER NOT NULL PRIMARY KEY GENERATED always AS 
     IDENTITY (START WITH 1, 
     INCREMENT BY 1), 
     sim_name                   VARCHAR(510) NOT NULL, 
     model_name                 VARCHAR(510) NOT NULL, 
     exp_name                   VARCHAR(510) NOT NULL, 
     exp_start_time_stamp       TIMESTAMP, 
     exp_end_time_stamp         TIMESTAMP, 
     num_reps                   INTEGER NOT NULL CHECK (num_reps >=1), 
     last_rep                   INTEGER, 
     length_of_rep              DOUBLE, 
     length_of_warm_up          DOUBLE, 
     has_more_reps              BOOLEAN, 
     rep_allowed_exec_time      BIGINT, 
     rep_init_option            BOOLEAN, 
     reset_start_stream_option  BOOLEAN, 
     antithetic_option          BOOLEAN, 
     adv_next_sub_stream_option BOOLEAN, 
     num_stream_advances        INTEGER 
  )

**Comment by lukaseder** [comment]
Thanks a lot for reporting. Curious: What database dialect allows (requires?) a comma between the `START WITH` and `INCREMENT BY` clauses?

**Comment by rossetti** [comment]
Derby.  I was just following this documentation:

https://db.apache.org/derby/docs/10.0/manuals/reference/sqlj27.html#CREATE+TABLE+statement
https://db.apache.org/derby/docs/10.0/manuals/develop/develop132.html

However, I ran it on a derby database with and without the comma and the SQL executed without any errors. So, it appears as you suggest, that the comma is optional, but the derby documentation makes it look like it is not.

**Comment by lukaseder** [comment]
Thanks a lot for the link. Interesting. We'll parse it as optional for all databases. Fixed for jOOQ 3.12. Backport to 3.11.3 scheduled: #7636
